engine/resources/user.go

```diff
@@ -411,6 +431,7 @@ func (obj *UserRes) UIDs() []engine.ResUID {
 	x := &UserUID{
 		BaseUID: engine.BaseUID{Name: obj.Name(), Kind: obj.Kind()},
 		name:    obj.Name(),
+		uid:     obj.UID,
```

Comment:
It's not used in this PR but I imagine this will be desirable in the future.
I think it would be reasonable for any resource that takes a `User string` to attempt to `strconv.ParseInt` and treat that username as a uid if it's a valid UID. I suppose alternatively resources could add a User/UID `union` and only allow one or the other